backend-api/src/functions/jwks/jwksBuilder/jwksBuilder.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  GetPublicKeyCommand,
+  GetPublicKeyCommandOutput,
+  KMSClient,
+} from "@aws-sdk/client-kms";
+import { NodeHttpHandler } from "@aws-sdk/node-http-handler";
+import { errorResult, Result, successResult } from "../../utils/result";
+import { createPublicKey } from "node:crypto";
+import {
+  EncryptionJwk,
+  EncryptionJwkAlgorithm,
+  EncryptionJwkUse,
+  Jwks,
+} from "../../types/jwks";
+
+export class JwksBuilder implements IJwksBuilder {
+  constructor(
+    private readonly keyId: string,
+    private readonly kmsClient = new KMSClient([
+      {
+        region: "eu-west-2",
+        maxAttempts: 2,
+        requestHandler: new NodeHttpHandler({
+          connectionTimeout: 29000,
+          requestTimeout: 29000,
+        }),
+      },
+    ]),
+  ) {}
+
+  async buildJwks(): Promise<Result<Jwks>> {
+    const jwks: Jwks = {
+      keys: [],
+    };
+    const result = await this.getPublicKeyAsJwk();
+    if (result.isError) {
+      return result;
+    }
+
+    jwks.keys.push(result.value);
+    return successResult(jwks);
+  }
+
+  async getPublicKeyAsJwk(): Promise<Result<EncryptionJwk>> {
+    let getPublicKeyOutput: GetPublicKeyCommandOutput;
+    try {
+      const command = new GetPublicKeyCommand({
+        KeyId: this.keyId,
+      });
+      getPublicKeyOutput = await this.kmsClient.send(command);
+    } catch {
+      return errorResult({
+        errorMessage: "Error from KMS",
```

Comment:
Should we make this more specific to what's going on here or say it's unexpected?